Actions/DetermineDeploymentEnvironments/DetermineDeploymentEnvironments.ps1

```diff
@@ -0,0 +1,177 @@
+$environments = @($ghEnvironments | ForEach-Object { $_.name }) + @($settings.environments) | Select-Object -unique | Where-Object { $settings.excludeEnvironments -notcontains $_ -and $_ -like $getEnvironments }
+
+Write-Host "Environments found: $($environments -join ', ')"
+
+$deploymentEnvironments = @{}
+
+if (!($environments)) {
+    # If no environments are defined and the user specified a single environment, use that environment
+    # This allows the user to specify a single environment without having to define it in the settings
+    if ($getenvironments -notcontains '*' -and $getenvironments -notcontains '?' -and $getenvironments -notcontains ',') {
+        $envName = $getEnvironments.Split(' ')[0]
+        $deploymentEnvironments += @{
+            "$getEnvironments" = @{
+                "EnvironmentName" = $envName
+                "Branches" = $null
+                "BranchesFromPolicy" = $null
+                "Projects" = '*'
+                "ForceSync" = $false
+                "ContinuousDeployment" = !($getEnvironments -like '* (PROD)' -or $getEnvironments -like '* (Production)' -or $getEnvironments -like '* (FAT)' -or $getEnvironments -like '* (Final Acceptance Test)')
+                "runs-on" = @($settings."runs-on".Split(',').Trim())
+            }
+        }
+    }
+}
+else {
+    $environments | ForEach-Object {
+        $environmentName = $_
+        Write-Host "Environment: $environmentName"
+        $envName = $_.Split(' ')[0]
+
+        # Check Obsolete Settings
+        "$($envName)-Projects","$($envName)_Projects" | ForEach-Object {
+            if ($settings.Contains($_)) {
+                throw "The setting $_ is obsolete and should be replaced by using the Projects property in the DeployTo$envName setting in .github/AL-Go-Settings.json instead"
+            }
+        }
+
+        # Default Deployment settings are:
+        # - environment name: same
+        # - branches: main
+        # - projects: all
+        # - continuous deployment: only for environments not tagged with PROD or FAT
+        # - runs-on: same as settings."runs-on"
+        $deploymentSettings = @{
+            "EnvironmentType" = "SaaS"
+            "EnvironmentName" = $envName
+            "Branches" = $null
+            "BranchesFromPolicy" = $null
+            "Projects" = '*'
+            "ForceSync" = $false
+            "ContinuousDeployment" = $null
+            "runs-on" = @($settings."runs-on".Split(',').Trim())
+        }
+
+        # Check DeployTo<environmentName> setting
+        $settingsName = "DeployTo$envName"
+        if ($settings.ContainsKey($settingsName)) {
+            # If a DeployTo<environmentName> setting exists - use values from this (over the defaults)
+            $deployTo = $settings."$settingsName"
+            'EnvironmentType','EnvironmentName','Branches','Projects','ForceSync','ContinuousDeployment','runs-on' | ForEach-Object {
+                if ($deployTo.ContainsKey($_)) {
```

Comment:
Yeah, it is copying properties to a HashTable, but MergeCustomObject actually does more than this - it will also merge arrays and other things rules that applies to having multiple settings files.
This one is overriding a default set of settings with calculated settings from ReadSettings - I would like to keep it simple.

We could move the defaulting of the DeployTo structure to ReadSettings (I thought about that) but we would also need the defaulting here, as we don't know the environments in ReadSettings and if the DeployTo key doesn't exist we cannot default it. Keeping defaulting of DeployTo structure here makes more sense IMO.